### DISCUSSION TITLE
Auto-route Satin Stitch Undo?

### DISCUSSION THREAD

**Discussion by aih777** [inkstitch/inkstitch#1557] [Q&A]
I've been playing around with the auto-route satin stitch feature in order to reduce workload when designing lettering for patches, but I just realized on a current design that id like to change the way the satin stitches are routed to hide the jump stitches under the next layer of color.

Basically what I am asking is if there is a way to undo auto-routing without using ctrl+z (I have made too much progress since using auto-route). I notice that Auto-route has left behind the old group names of each satin letter in the objects panel so was wondering if this was already possible. I have already consulted the inkstitch webpage, the wiki, this github, and a few YouTube videos and have not found anything about undoing the process.

If not, perhaps I can redo the lettering from scratch, but would like to avoid this and hopefully pick up some more tricks that I can use for later designs.

**Comment by tyrosinase** [comment]
FWIW, I have done some massive undo chains: you can pause the undo, copy something you do want to preserve into another document (or just onto the clipboard if you're brave), and keep undoing and then paste the changes you want to keep back in. Or copy things that got wiped out, then do a redo back to the current time and restore them manually.

This doesn't help now, but I've also been in the habit of keeping an un-routed version of things in a hidden layer.

It doesn't re-join segments that auto-routing has split, but you can just select one of the positioning stitches, right-click and Select Same -> Stroke Style, and (probably) delete only the auto-route positioning.

**Reply by aih777** [reply]
> FWIW, I have done some massive undo chains: you can pause the undo, copy something you do want to preserve into another document (or just onto the clipboard if you're brave), and keep undoing and then paste the changes you want to keep back in. Or copy things that got wiped out, then do a redo back to the current time and restore them manually.
> 
> This doesn't help now, but I've also been in the habit of keeping an un-routed version of things in a hidden layer.
> 
> It doesn't re-join segments that auto-routing has split, but you can just select one of the positioning stitches, right-click and Select Same -> Stroke Style, and (probably) delete only the auto-route positioning.

Ah, I didn't think about that, thank you, I will definitely start to do those things for my next designs. Especially keeping an un-routed version on a separate layer, that will definitely save me a lot of headache if [read: when] i next mess up.
